scheduler/src/cook/mesos/dru.clj

```diff
@@ -64,19 +64,38 @@
                           task-resources)]
     scored-tasks))
 
-(defn scored-task->priority
-  "The higher the dru of task, the lower the priority"
-  [scored-task]
-  (- (:dru scored-task)))
+(defn sorted-merge
+  "Accepts a seq-able datastructure `colls` where each item is seq-able.
+   Each seq-able in `colls` is assumed to be sorted based on `key-fn`
+   Returns a lazy seq containing the sorted items of `colls`
+
+   Initial code from: http://blog.malcolmsparks.com/?p=42"
+  ([key-fn ^java.util.Comparator comp-fn colls]
+   (letfn [(next-item [[_ colls]]
+             (if (nil? colls)
+               [:end nil] ; Allow nil items in colls
+               (let [[[yield & remaining] & other-colls]
+                     (sort-by (comp key-fn first) comp-fn colls)]
+                 [yield (if remaining (cons remaining other-colls) other-colls)])))]
```

Comment:
it's a bit strange to use cons instead of conj here (since we don't care where it's added?), but I guess it doesn't matter